tests/vm_watch_test.go

```diff
@@ -0,0 +1,330 @@
+		}
+	}
+
+	// Reads from stdin until a newline character is found
+	readLine := func(rc io.ReadCloser, timeout time.Duration) string {
+		lineChan := make(chan string)
+
+		go func() {
+			defer GinkgoRecover()
+
+			var line strings.Builder
+			buf := make([]byte, 1)
+			defer close(lineChan)
+
+			for {
+				n, err := rc.Read(buf)
+
+				if err != nil && err != io.EOF && !strings.Contains(err.Error(), "file already closed") {
+					Fail(err.Error())
+					return
+				}
+
+				if n > 0 {
+					if buf[0] != '\n' {
+						line.WriteByte(buf[0])
+					} else {
+						break
+					}
+				}
+			}
+
+			lineChan <- line.String()
+		}()
+
+		select {
+		case line := <-lineChan:
+			return line
+		case <-time.After(timeout):
+			err := rc.Close()
+			Expect(err).ToNot(HaveOccurred(), "stdout should have been closed properly")
+
+			Fail("Timeout reached on read operation")
+
+			return ""
+		}
+	}
+
+	// Reads VM status from the given pipe (stdin in this case) and
+	// returns a new status.
+	// if old_status is non-nil, the function will read status lines until
+	// new_status.running != old_status.running in order to skip duplicated status lines
+	readVMStatus := func(rc io.ReadCloser, old_status *vmStatus, timeout time.Duration) *vmStatus {
+		new_stat := newVMStatus(strings.Fields(readLine(rc, timeout)))
+
+		for old_status != nil && new_stat.running == old_status.running {
+			new_stat = newVMStatus(strings.Fields(readLine(rc, timeout)))
+		}
+
+		return new_stat
+	}
+
+	// Reads VMI status from the given pipe (stdin in this case) and
+	// returns a new status.
+	// if old_status is non-nil, the function will read status lines until
+	// new_status.phase != old_status.phase in order to skip duplicated lines
+	readVMIStatus := func(rc io.ReadCloser, old_status *vmiStatus, timeout time.Duration) *vmiStatus {
+		newStat := newVMIStatus(strings.Fields(readLine(rc, timeout)))
+
+		for old_status != nil && newStat.phase == old_status.phase {
+			newStat = newVMIStatus(strings.Fields(readLine(rc, timeout)))
+		}
+
+		return newStat
+	}
+
+	// Create a command with output/error redirection.
+	// Returns (cmd, stdout, stderr)
+	createCommandWithNSAndRedirect := func(namespace, cmdName string, args ...string) (*exec.Cmd, io.ReadCloser, io.ReadCloser) {
+		cmdName, cmd, err := tests.CreateCommandWithNS(namespace, cmdName, args...)
+
+		Expect(cmdName).ToNot(Equal(""))
+		Expect(cmd).ToNot(BeNil())
+		Expect(err).ToNot(HaveOccurred(), "Command should have been created with proper kubectl/oc arguments")
+
+		// Output redirection
+		stdOut, err := cmd.StdoutPipe()
+		Expect(err).ToNot(HaveOccurred(), "stdout should have been redirected")
+		Expect(stdOut).ToNot(BeNil())
+
+		stdErr, err := cmd.StderrPipe()
+		Expect(err).ToNot(HaveOccurred(), "stderr should have been redirected")
+		Expect(stdErr).ToNot(BeNil())
+
+		return cmd, stdOut, stdErr
+	}
+
+	It("[test_id:3468]Should update vm status with the proper columns using 'kubectl get vm -w'", func() {
```

Comment:
@omeryahud the reason for failure here can be that the PR for this might not have made it back to previous versions.
https://github.com/kubernetes/kubernetes/pull/76161

Need to check specific version, but I think it should have made in 1.16 and onwards
You should skip the tests for the versions before that.

Reply:
Thanks, working on it now